src/upchunk.ts

```diff
@@ -290,20 +390,46 @@ export class UpChunk {
     this.success = false;
     this.nextChunkRangeStart = 0;
 
+    if (options.useFileSliceFallback) {
```

Comment:
This basically monitors for an error event and then checks if there is also a corresponding error on the current instance of `this.chunkedIterable`. We could instead add concepts like error types/categories info to the event itself if this feels a bit too "acrobatics-y". Larger refactors to capture this logic in `sendChunks()` felt like the wrong  path forward, especially if we want to treat this as a "stopgap" as much as possible.

Reply:
Explicitly calling it a workaround may help us/contributors remember why it's like this. How would you feel about `useLargeFileWorkaround` or something?